examples/detection_example.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include "../include/feature_lib.h"
+
+std::vector<uint8_t> matToVector(const cv::Mat &mat) {
+  std::vector<uint8_t> result;
+  cv::Mat cont = mat.isContinuous() ? mat : mat.clone();  // make sure the Mat is continuous
+  cont.reshape(1, 1).copyTo(result);  // copy from a 1D Mat
+  return result;
+}
+
+// Calculates key points and image descriptors
+// Takes a filepath as input
+// Returns data in the structure declared in associated header
```

Comment:
Ir actually returns nothing, it prints keypoints and descriptors to `stdout`

Reply:
done